crates/sui-indexer-alt-jsonrpc/tests/write_tests.rs

```diff
@@ -0,0 +1,161 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::net::{IpAddr, Ipv4Addr, SocketAddr};
+
+use anyhow::Context;
+use prometheus::Registry;
+use reqwest::Client;
+use serde_json::{json, Value};
+use sui_indexer_alt_jsonrpc::{
+    config::RpcConfig, data::system_package_task::SystemPackageTaskArgs, start_rpc, RpcArgs,
+};
+use sui_pg_db::{
+    temp::{get_available_port, TempDb},
+    DbArgs,
+};
+use sui_swarm_config::genesis_config::AccountConfig;
+use test_cluster::TestClusterBuilder;
+use tokio::task::JoinHandle;
+use tokio_util::sync::CancellationToken;
+use url::Url;
+
+const EPOCH_DURATION_MS: u64 = 2000;
+const GAS_OBJECT_COUNT: usize = 1;
+const DEFAULT_GAS_AMOUNT: u64 = 1_000_000_000;
+
+#[tokio::test]
+async fn test_execution_and_dry_run() {
```

Comment:
Let's add some tests for various error cases, and make sure the error codes get reported faithfully.